### PR TITLE
fix: App crashes in User Profile Section #729

### LINKED ISSUE #729: App crashes in User Profile Section
**Summary:**

App crashes on opening User Profile Section

**Steps to reproduce:**

Open User Profile from Navigation Drawer by clicking on Personal Image

**Expected behaviour:**

App should not crash

**Observed behaviour:**

App crashes

**Device and Android version:**

Xiaomi

**Screenshots:**

issue_user_profile_crashes

### PR DESCRIPTION
**Pull request by ankurs287**
Fixes #729 

Reason behind crash:
1) When there is no entry in database for client's group. 
2) When there is no entry in database for client's dob.

### CODE REVIEW

**Comment by miPlodder** [comment]
@ankurs287 When working on issues which is assigned to someone else, you are first requested to ask other Contributor, if he/she is working or not. 

Please follow the Contribution guidelines from next time,
https://github.com/openMF/mifos-mobile/blob/development/.github/CONTRIBUTING.md

**Comment by ankurs287** [comment]
@therajanmaurya  @dilpreet96  @droidchef @satyan  I've fixed #729 . Kindly, look into it. Thanks.

**Comment by ankurs287** [comment]
@miPlodder Hey! This is my first time contributing to open-source projects. Sorry if I violated some points.

**Comment by miPlodder** [comment]
@ankurs287 Not a problem, but Please follows the guidelines in future. 

Btw Welcome to Open Source and keep contributing 👍

File: app/src/main/java/org/mifos/mobilebanking/ui/fragments/UserProfileFragment.java [hunk 1]
```diff
@@ -167,7 +167,8 @@ private String getGroups(List<Group> groups) {
         for (Group group : groups) {
             builder.append(getString(R.string.string_and_string, group.getName(), " | "));
         }
-        return builder.toString().substring(0, builder.toString().length() - 2);
+        if (groups.size() == 0) return ""; // no groups entry in database for the client
```
Comment (therajanmaurya) [inline app/src/main/java/org/mifos/mobilebanking/ui/fragments/UserProfileFragment.java]: Use if else here like this
```
 if (groups.size() == 0) {
      return "";
} else {
     return builder.toString().substring(0, builder.toString().length());
}
```

File: app/src/main/java/org/mifos/mobilebanking/utils/DateHelper.java [hunk 2]
```diff
@@ -26,11 +27,11 @@
 
     /**
      * the result string uses the list given in a reverse order ([x, y, z] results in "z y x")
-     *
      * @param integersOfDate [year-month-day] (ex [2016, 4, 14])
      * @return date in the format day month year (ex 14 Apr 2016)
      */
     public static String getDateAsString(List<Integer> integersOfDate) {
+        if (integersOfDate.size() != 3) return ""; // no data entry in database for the client
```
Comment (therajanmaurya) [inline app/src/main/java/org/mifos/mobilebanking/utils/DateHelper.java]: No need to add check here, Please remove it. Where ever you are trying to parse this date. use if lese there 

```
if (integersOfDate.size() != 3) {
// No DOB Found 
} else {
// Show DOB after parsing.
}
```

**Review by therajanmaurya — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/mifos/mobilebanking/ui/fragments/UserProfileFragment.java (hunk 1 above)
Comment (ankurs287) [reply app/src/main/java/org/mifos/mobilebanking/ui/fragments/UserProfileFragment.java]: okay. I will do that.

File: app/src/main/java/org/mifos/mobilebanking/utils/DateHelper.java (hunk 2 above)
Comment (ankurs287) [reply app/src/main/java/org/mifos/mobilebanking/utils/DateHelper.java]: okay.

**Comment by ankurs287** [comment]
@therajanmaurya new changes have been made. Kindly look into that. Thanks.

**Review by therajanmaurya — APPROVED** [review]

**Comment by therajanmaurya** [comment]
Fixed in https://github.com/openMF/mifos-mobile/pull/757

**Pull request closed without merging**